### DISCUSSION TITLE
Toggle colorscheme with vim opened

### DISCUSSION THREAD

**Discussion by gonstoll** [wezterm/wezterm#4612] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

When trying to toggle the color scheme with `'toggle-colorscheme'` event and Neovim opened, the theme toggles but only the statusbar is changed 🤔

```lua
wezterm.on('toggle-colorscheme', function(window)
  local overrides = window:get_config_overrides() or {}

  if not overrides.color_scheme then
    overrides.color_scheme = dark_theme
  else
    overrides.color_scheme = overrides.color_scheme == light_theme and dark_theme or light_theme
  end

  window:set_config_overrides(overrides)
end)
```

Take the following recording. When I open neovim, have a look at the bottom statusline bar:

https://github.com/wez/wezterm/assets/25900010/64e62d9e-e269-4271-963f-bcfb7b6e578f

Has someone managed to get Neovim to actually listen for this change and have `vim.o.background` updated?

**Comment by gegoune** [comment]
It's because you need to react to theme change from vim itself. You will need to change your `colorscheme` (or just `background`) option. I use https://github.com/cormacrelf/dark-notify to feed that information back to Neovim. Not sure how to do that in vim, sorry.

**Reply by gonstoll** [reply]
Was aware of `dark-notify`, but that only works with system preferences theme change, right?

**Reply by gegoune** [reply]
That's true. I am not sure if vim can receive remote command, but if it does you can send it some signal/message informing about change of western's colorcheme.

**Comment by wez** [comment]
FWIW, you can dodge a lot of complexity by configuring your vim color scheme to use the terminal color scheme and transparency by using good old fashioned ansi palette indices.

That way you can centralize the terminal theme and responsivity to dark mode stuff in your wezterm config, and things will work correctly even if you are logged into a remote system with no gui of its own, and no way for it to know about the appearance of your local desktop.

As an example, this is my personal vim color scheme that I've used for many many years:
https://gist.github.com/wez/afea18dba82825663712c882947e4aa4
There's some light discussion around this in https://www.reddit.com/r/neovim/comments/16lwnga/identify_wezterm_vim_color_schemes/
